train_tokenizer.py

```diff
@@ -0,0 +1,162 @@
+        trainer=trainer,
+        length=len(entire_dataset["train"]))
+
+    # trim_offsets=False tells post-processor to keep spaces as part of tokens
+    tokenizer.post_processor = processors.TemplateProcessing(
+        single="<bos> $A",
+        special_tokens=[("<bos>", tokenizer.token_to_id("<bos>"))],
+    )
+
+    # Add decoder for converting tokens back to text
+    tokenizer.decoder = decoders.ByteLevel()
+
+    # Enable padding
+    tokenizer.enable_padding(
+        direction="right",
+        pad_id=0,
+        pad_token="<pad>",
+        length=seq_len + 1)
+
+    # Enable truncation
+    tokenizer.enable_truncation(max_length=seq_len + 1, direction="right")
+
+    # Wrap tokenizer with transformers library
+    tokenizer = PreTrainedTokenizerFast(
+        model_max_length=seq_len,
+        padding_side="right",
+        truncation_side="right",
+        bos_token="<bos>",
+        unk_token="<unk>",
+        pad_token="<pad>",
+        tokenizer_object=tokenizer)
+
+    # Save tokenizer to file
+    tokenizer_save_path = Path(tokenizer_folder) / dataset_name
+    tokenizer_save_path.mkdir(parents=True, exist_ok=True)
+    tokenizer.save_pretrained(tokenizer_save_path)
+
+if __name__ == "__main__":
+    # Get arguments
+    parser = ArgumentParser()
+
+    parser.add_argument(
+        "--tokenizer_folder",
+        type=str,
+        required=True,
+        help="Folder to save tokenizer to.")
+    parser.add_argument(
+        "--dataset_name",
+        type=str,
+        required=True,
+        help="Name of Hugging Face dataset.")
+    parser.add_argument(
+        "--seq_len",
+        type=int,
+        required=True,
+        help="Context window/sequence length.")
+    parser.add_argument(
+        "--vocab_size",
+        type=int,
+        required=True,
+        help="Maximum vocabulary size.")
+    parser.add_argument(
+        "--dataset_dir",
+        type=str,
+        required=True,
+        help="Relative path from base of repository to directory in which to download the dataset.")
+    parser.add_argument(
+        "--dataset_subset",
+        type=str,
+        default=None,
+        help="Configuration/subset of dataset to use.")
+    parser.add_argument(
+        "--text_feature",
+        type=str,
+        default="text",
+        help="Name of the feature/column of the dataset to use.")
+    parser.add_argument("--splits", type=float, nargs=3,
+        default=[0.7, 0.2, 0.1],
+        help="Space-separated decimal splits of train, validation, and " + \
+            "test datasets. (Ex: '0.7 0.2 0.1')")
+    parser.add_argument(
+        "--rand_seed",
+        type=int,
+        default=None,
+        help="Seed used during dataset shuffling, ignored if None.")
+
+    args = parser.parse_args()
+
+    train_tokenizer(args.tokenizer_folder, args.dataset_name, args.seq_len, args.vocab_size, args.dataset_dir, args.dataset_subset, args.text_feature, args.splits, args.rand_seed)
```

Comment:
This is fine, but please look at train_model.py and grid_search.py for more effective way to pass in args -->  train_model(**vars(args))
**vars(args) is simple and works.